File: 2026-03/src/day_19/boj_11501/Main.cpp
#include <bits/stdc++.h>
using namespace std;

int arr[1000000];

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    int t;
    cin >> t;

    while (t--) {
        int n;
        cin >> n;

        for (int i = 0; i < n; i++) cin >> arr[i];

        long long ans = 0;
        int mx = 0;
        for (int i = n - 1; i >= 0; i--) {
            mx = max(mx, arr[i]);
            ans += mx - arr[i];
        }

        cout << ans << '\n';
    }
}

File: 2026-03/src/day_19/boj_11501/Main.java
package day_19.boj_11501;

import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        StringTokenizer st;

        int t = Integer.parseInt(br.readLine());
        while (t-- > 0) {
            int n = Integer.parseInt(br.readLine());

            int[] arr = new int[n];
            st = new StringTokenizer(br.readLine());
            for (int i = 0; i < n; i++) {
                arr[i] = Integer.parseInt(st.nextToken());
            }

            long ans = 0;
            int max = 0;
            for (int i = n - 1; i >= 0; i--) {
                max = Math.max(max, arr[i]);
                ans += max - arr[i];
            }

            sb.append(ans).append("\n");
        }

        System.out.println(sb);
    }
}
